grafonnet-base/util/panel.libsonnet

```diff
@@ -13,22 +13,28 @@ local d = import 'github.com/jsonnet-libs/docsonnet/doc-util/main.libsonnet';
     |||,
     args=[
       d.arg('panels', d.T.array),
-      // d.arg('init', d.T.number) // intentionally undocumented, for internal use
     ]
   ),
-  setPanelIDs(panels, init=0): [
-    panels[i - 1] { id: i + init }
-    + (if panels[i - 1].type == 'row'
-          && 'panels' in panels[i - 1]
-       then {
-         panels:
-           this.setPanelIDs(
-             panels[i - 1].panels,
-             init=(std.length(panels) * i)
-           ),
-       }
-       else {})
-    for i in std.range(1, std.length(panels))
-  ],
+  setPanelIDs(panels):
+    local indexStart = 1;
+    local infunc(panels, init) =
+      std.mapWithIndex(
+        function(index, panel)
+          panel { id: indexStart + index + init }
+          + (
+            if panel.type == 'row'
+               && 'panels' in panel
+            then {
+              panels:
+                infunc(
+                  panel.panels,
+                  std.length(panels) * (indexStart + index)
```

Comment:
Could this cause collision? 🤔 

For instance, if we have 1 panel plus 2 row panels with 10 panels each:
- The first panel will be identified with `1`
- The panels inside the first row will be identified from `3 * (1 + 1)`: 6, 7, 8, 9, 10... 
- The panels inside the second row will be identified from `3 * (1 + 2)`: 9, 10, 11, 12...

Or am I missing something?

PS: For some reason, I completely missed that during the mob session 🙏🏻

Reply:
I'm also wondering about collisions :thinking: 

Is there a reason why we don't use a sequential ID for the panels?

Reply:
I believe you both are right. This algorithm works for some cases but not all, it definitely comes up when having a lot of panels in a row.

This needs different logic. There is no particaluar reason not to use sequential IDs.